src/react/reconcilation.js

```diff
@@ -153,8 +153,10 @@ export class Reconciler {
     let value;
     let childContext = context;
 
-    // first we check if it's a class component
-    if (valueIsClassComponent(this.realm, componentType)) {
+    // first we check if it's a legacy class component
+    if (valueIsLegacyCreateClassComponent(this.realm, componentType)) {
+      throw new ExpectedBailOut("components created with create-react-class are not supported");
```

Comment:
I was wondering about this pattern of embedding user facing error messages inside exceptions, so I had a closer look and I noticed that you promote these to PP0019 at some point. That undocumented error code has now been grabbed by someone else, so please change it and document it. While you are at it, also consider making a separate error code (with specialized documentation) for each instance of ExpectedBailOut.

Also, if you wan't to fit in with best practice (and why not) then report the diagnostic at the point where you do the check and then bail out using FatalError (unless the error is recoverable, in which case the error handler can tell you to repair and carry on).

Reply:
That's strange, I've been using `PP0019` for some time now and documented on the wiki page. 

In regards to your other comments. They only get promoted if the root component in the reconciliation tree bails out; if components further down the tree bail-out, it skips over them and continues (but adds an inline message). I didn't want to overload the details of `ExpectedBailOut` by having each one give a code, as this shouldn't be the case. Maybe I misunderstand you here.

Reply:
Looking at your documentation for the error code it seems to me that is essentially just saying "hey something is wrong with code, go figure it out yourself". If you want the documentation to be a bit more helpful, the error has to be more specific to a particular situation and there has to be a useful hint about how the error may be avoided.

Reply:
@hermanventer You should see the message in the DiagnosticError, as the errors are all expected ones (other errors get passed through), the message should be clear as to what happened to cause the problem.